fml/time/time_recorder.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_TIME_TIME_RECORDER_H_
+#define FLUTTER_FML_TIME_TIME_RECORDER_H_
+
+#include <mutex>
+#include "flutter/fml/time/time_point.h"
+
+namespace fml {
+
+/// Utility class to record time in a thread-safe manner.
```

Comment:
I am not sure we needed such a specific utility for something that is relatively simple (lock around an std::optional<T>). Are you going to add to this in the future? Maybe this can be called:

```
template<class T>
class ThreadSafeRecorder ...
```

There isn't really anything time/chrono related here. You just happen to use it for a `fml::TimePoint`.